trackintel/model/util.py

```diff
@@ -27,33 +27,26 @@ class TrackintelGeoDataFrame(GeoDataFrame):
     Helper class to subtype GeoDataFrame correctly and fallback to custom class.
 
     Three possible outcomes for the _constructor
-    - If check fails or is DataFrame without fallback_class set, return GeoDataFrame/DataFrame instance w/o changes
-    - If check succeeds and has geometry, change class to current class and return
-    - If check succeeds and has no geometry, change class to fallback_class and return
+    - If GeoDataFrame, keep it a TrackintelGeoDataFrame
+    - If DataFrame without fallback class, just return the DataFrame
+    - If DataFrame with fallback class, fall back to it.
     """
 
-    # set fallback_class for succeeding _check and missing geometry, if None default to pd.DataFrame
+    # set fallback_class for missing geometry, if None default to pd.DataFrame
     fallback_class = None
 
-    @staticmethod
-    def _check(self, validate_geometry=True):
-        raise NotImplementedError
-
     @property
     def _constructor(self):
         """Interface to subtype pandas properly"""
         super_cons = super()._constructor
 
         def _constructor_with_fallback(*args, **kwargs):
             result = super_cons(*args, **kwargs)
-            if not self._check(result, validate_geometry=False):
-                # check fails -> not one of our classes
-                return result
             if isinstance(result, GeoDataFrame):
-                # cannot validate geometry as geometry column is maybe not set
-                return self.__class__(result, validate_geometry=False)
+                return self.__class__(result, validate=False)
+            # uses DataFrame constructor -> must be DataFrame
             if self.fallback_class is not None:
-                return self.fallback_class(result)
+                return self.fallback_class(result, validate=False)
```

Comment:
The idea is that we never escape the trackintel classes for trips.
Else one might encounter this behavior.
````python
tripsGDF = tripsGDF.drop(columns="user_id")
# some other code
tripsGDF.drop(columns="geometry")
````
This code would raise an error because `TripsDataFrame` expects a column "user_id", but the error producing line is maybe quite far away. So just setting `validate=False` avoids this and later the shape is tested manually with `validate`.